Jenkinsfile

```diff
@@ -491,6 +491,14 @@ pipeline {
 
                     . make/include/secrets
 
+                    # Get the last updated secret
+                    secret_resource="\$(kubectl get secrets --namespace="${jobBaseName()}-${BUILD_NUMBER}-scf" --output=jsonpath='{.items[-1:].metadata.name}' --sort-by=.metadata.resourceVersion)"
+
+                    # Get a random secret
+                    secret_name="\$(kubectl get secret --namespace="${jobBaseName()}-${BUILD_NUMBER}-scf" "\${secret_resource}" -o json | jq -r '.data | to_entries | .[].key | select(contains(".generator") | not) ' | sort --random-sort | head -n 1)"
```

Comment:
I don't think this will work because you don't exclude immutable secrets (which I don't think is possible without parsing the role manifest again).